### PR TITLE
Fix crash when selecting avatar in context tags

### LINKED ISSUE #13908: App crashes when selecting Avatar type context tag in quo2
## Summary
App crashes when selecting Avatar type context tag in quo2 

### Reproduction:

https://user-images.githubusercontent.com/17097240/187774805-8ad4d382-72b5-423e-8469-2ad3d5374efc.mp4

### Additional Information

- Status version: nightly

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #13908 
status: ready

### CODE REVIEW

**Comment by Parveshdhull** [comment]
Hi @ibrkhalil ,
Thank you very much for your work. The app doesn't crash anymore.
But changing contact is not affecting the avatar:

https://user-images.githubusercontent.com/17097240/188283928-0349d8f8-7f79-4838-b7c3-ad387fc915d6.MP4

File: src/quo2/components/context_tags.cljs [hunk 1]
```diff
@@ -47,8 +48,10 @@
       (trim-public-key public-key)]]))
 
 (defn user-avatar-tag [_ _]
-  (fn [params public-key]
-    (let [{:keys [name photo]} @(re-frame/subscribe [:contacts/name-and-photo public-key])]
+  (fn [params _]
+    (let [{:keys [names public-key]} @(re-frame/subscribe [:multiaccount/contact])
+          {:keys [nickname three-words-name]} names
```
Comment (Parveshdhull) [inline src/quo2/components/context_tags.cljs]: As components should be abstract without any reframe calls,
should we move this subscription to `quo2/screens/context_tags.cljs:51`?

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil , Thank you very much for your work. The app doesn't crash anymore. But changing contact is not affecting the avatar:
> 
>  RPReplay_Final1662230069.MP4

Will look into it .. Thanks Parvesh

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil , Thank you very much for your work. The app doesn't crash anymore. But changing contact is not affecting the avatar:
> 
>  RPReplay_Final1662230069.MP4

Totally missed that I should use data from user's contacts, I instead used currently logged in user data .. Fixed

File: src/quo2/components/context_tags.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/quo2/components/context_tags.cljs]: i think even in screens we shouldn't use re-frame, there must be just mock data

File: src/quo2/components/context_tags.cljs [hunk 2]
```diff
@@ -46,9 +46,15 @@
                  :size :paragraph-2}
       (trim-public-key public-key)]]))
 
-(defn user-avatar-tag [_ _]
-  (fn [params public-key]
-    (let [{:keys [name photo]} @(re-frame/subscribe [:contacts/name-and-photo public-key])]
+(defn user-avatar-tag [params public-key contacts]
+  (fn [params public-key contacts]
+    (let [photo @(re-frame.core/subscribe [:chats/photo-path public-key])
```
Comment (flexsurfer) [inline src/quo2/components/context_tags.cljs]: use mock data from screens instead

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/quo2/components/context_tags.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/quo2/components/context_tags.cljs]: Done

File: src/quo2/screens/context_tags.cljs [hunk 3]
```diff
@@ -12,38 +12,46 @@
...
-                     (map (fn [{:keys [public-key]}]
-                            {:key public-key
-                             :value (multiaccounts/displayed-name
-                                     (get contacts public-key))})
-                          (vals contacts))}))]
+            (cond
+              (= (:type @state) :group-avatar) (conj main-descriptor {:label "Label"
+                                                                      :key   :label
+                                                                      :type  :text})
+              (= (:type @state) :avatar) (let [photo @(re-frame.core/subscribe [:chats/photo-path (:contact @state)])]
```
Comment (flexsurfer) [inline src/quo2/screens/context_tags.cljs]: still better to not use re-frame in quo, even here, but what is the value of this subscription? can we mock it?

Comment (ibrkhalil) [reply src/quo2/screens/context_tags.cljs]: I had it as a base64 string but removed it .. I'll use it again

Comment (ibrkhalil) [reply src/quo2/screens/context_tags.cljs]: Done

File: src/quo2/components/context_tags.cljs [hunk 4]
```diff
@@ -46,16 +46,21 @@
                  :size :paragraph-2}
       (trim-public-key public-key)]]))
 
-(defn user-avatar-tag [_ _]
-  (fn [params public-key]
-    (let [{:keys [name photo]} @(re-frame/subscribe [:contacts/name-and-photo public-key])]
+(defn user-avatar-tag [params public-key photo contacts]
+  (fn [params public-key photo contacts]
```
Comment (Parveshdhull) [inline src/quo2/components/context_tags.cljs]: Hi, why do we need to pass all contacts here. why can't we just pass final name?

**Review by flexsurfer — APPROVED** [review]

File: src/quo2/components/context_tags.cljs (hunk 4 above)
Comment (ibrkhalil) [reply src/quo2/components/context_tags.cljs]: Done

Comment (ibrkhalil) [reply src/quo2/components/context_tags.cljs]: Please check Parvesh

**Review by Parveshdhull — APPROVED** [review]
Thanks for the fix, looks good

**Pull request merged**